### PR TITLE
Create CLI for managing ACLs

### LINKED ISSUE #4: create CLI for sinopia_acl
We want a CLI where the admin user can:

(so far)
- specify a group and see comprehensible results of which webIDs are in the group on the server and what permissions each has.  (requires #2)
- specify a group and a webID to add and see comprehensible results showing that the webID was added. (requires #3) ... could be done as a separate ticket.

npm modules such as `cli`, `minimist` (for options parsing), `clui` (for table output) may be useful

### PR DESCRIPTION
**Pull request by mjgiarlo**
# Includes

* Create CLI class for managing ACLs
* Parse command-line args and options using `commander`
* Add `bin/acl` executable to package
* Get jest coverage working (did not print coverage results before)
* Opportunistic tweaks to `package.json`:
  * Add mjgiarlo as contributor
  * Remove apparently defunct `main` entry
  * Run `eslint` on all .js files (future-proofing in case we add dirs)
  * Remove `--runInBand` arg from jest invocations (unnecessary AFAICT?)
  * Add `jest-integration` npm script, setting up near-term integration work

# Question

Due to the way `commander` is tested—using another process per a recipe found on Google—none of the coverage added in this branch is picked up by static analysis. :unamused: But it's there! We can handle this in one of three ways:

1. Don't check `src/cli/commander.js` for code coverage. Possibly defensible, since we shouldn't need to test the `commander` library in `sinopia_acl`.
1. Resign ourselves to the low overall coverage.
1. Use a different strategy for testing our `commander` usage.

I lean towards 1, for expediency's sake.

# Notes

**Note 1**: I chose to use `commander` for the CLI since it's both featureful and maintained unlike some of the other libraries mentioned in #4.

**Note 2**: I considered using `npm run` instead of `bin/...` for this, but so far I'm preferring the `bin/...` approach to distinguish it from the developer experience of running stuff with `npm run`.

# Design

Here's how the CLI looks from the user perspective.

Print help if no commands or options provided:

```shell
$ bin/acl
Usage: acl [options] [command]

Options:
  -v, --version          output the version number
  -h, --help             output usage information

Commands:
  users <group>          List users and access controls in a group
  groups                 List all groups
  add <user> <group>     Add user to group
  remove <user> <group>  Remove user from group
```

Print help when requested (also works with `-h`):

```shell
bin/acl --help
Usage: acl [options] [command]

Options:
  -v, --version          output the version number
  -h, --help             output usage information

Commands:
  users <group>          List users and access controls in a group
  groups                 List all groups
  add <user> <group>     Add user to group
  remove <user> <group>  Remove user from group
```

Print version of acl tool (taken from package.json):

```shell
$ bin/acl -v
0.0.1
```

Show all groups:

```shell
$ bin/acl groups
not implemented yet
```

Throw an error when required argument is missing:

```shell
$ bin/acl users
error: missing required argument `group'
```

List all users in a group:

```shell
$ bin/acl users http://foo.bar
list() called with http://foo.bar
[ 'http://sinopia.io/users/cmharlow' ]
```

Add a user to a group:

```shell
$ bin/acl add userURI groupURI
adding userURI to groupURI
not implemented yet
```

Remove a user from a group:

```shell
$ bin/acl remove userURI groupURI
removing userURI to groupURI
not implemented yet
```

fixes #4

### CODE REVIEW

File: src/cli/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    this.acl = new WebAccessControl(fs.readFileSync('__tests__/__fixtures__/cmharlowBaseAcl.ttl', 'utf8'))
+  }
+
+  listUsers(groupUri) {
+    console.log(`list() called with ${groupUri}`)
+    console.log(this.acl.listUsers())
+  }
+
+  listGroups() {
+    console.log('not implemented yet')
```
Comment (ndushay) [inline src/cli/index.js]: note that this would be using sinopia_client code, theoretically

File: .codeclimate.yml [hunk 2]
```diff
@@ -42,6 +42,8 @@ checks:
 plugins:
   eslint:
     enabled: true
+    config:
+      config: .eslintrc.js
```
Comment (ndushay) [inline .codeclimate.yml]: a ha!  didn't realize this.

File: .eslintrc.js [hunk 3]
```diff
@@ -23,8 +23,8 @@ module.exports = {
   },
   overrides: [
     {
-      "files": ["src/*.js",
-                "__tests__/*.js"],
+      "files": ["src/**/*.js",
+                "__tests__/**/*.js"],
```
Comment (ndushay) [inline .eslintrc.js]: yeah, i meant to do that.

**Comment by mjgiarlo** [comment]
> I agree with option 1. Feel free to merge it yourself or to deal with coverage using option 1 and then merge it.

I will do the latter. Pls hold off on merging.

**Comment by mjgiarlo** [comment]
Chatted with @ndushay about the most recent commit in this branch and she's OK with my merging this given her prior approval. Merging now.

**Pull request merged**